crates/matrix-sdk/src/widget_api/api/mod.rs

```diff
@@ -68,7 +68,7 @@ fn process_message(
     req_sink: Sender<FromWidgetAction>,
     resp_sink: Sender<ToWidgetAction>,
 ) -> Result<()> {
-    match from_json(&raw).map_err(|_| Error::InvalidJSON)? {
+    match from_json(&raw).map_err(|e| Error::InvalidJSON(e.to_string()))? {
```

Comment:
We could define a `ParsingError(serde::Error)` and add a `#[from]` conversion to the error instead (since it's a typical error in our case).